File: src/08.dbpointer/15.643.ts
export function findMaxAverage(nums: number[], k: number): number {
    let res = 0;
    let sum = 0;

    for (let i = 0; i < k; i++) {
        sum += nums[i];
    }

    res = sum;

    for (let i = k; i < nums.length; i++) {
        sum = sum + nums[i] - nums[i - k];
        res = Math.max(res, sum);
    }

    return res / k;
}

File: src/08.dbpointer/15.643.go
package dbpointer

func FindMaxAverage(nums []int, k int) float64 {
	res := 0
	sum := 0
	for i := 0; i < k; i++ {
		sum += nums[i]
	}
	res = sum

	for i := k; i < len(nums); i++ {
		sum = sum + nums[i] - nums[i-k]
		if sum >= res {
			res = sum
		}
	}

	return float64(res) / float64(k)
}
